**Question/need:** Can we predict the success (gross revenues, revenues per budget dollar, and/or oscar nominations) of proposed low-budget films (< $10 million) based on its characteristics? Are there different predictive characteristics for success in terms of gross revenues vs. oscar nominations? 

**Movie data:** I'm planning on using all movies (1980 - 2016) from boxofficemojo to investigate the impact of various characteristics on success. If I have time, I would also like to incorporate other variables, such as google searches and wikipedia page views, as well as critic and general public rating information from rottentomatoes. 

**Characteristics of each movie and/or other entities:** I'd like to investigate as many characteristics as possible during my first pass to determine which ones have the greatest predictive impact, and then dig in deeper to these characteristics. The features I'm most interested in are: genre, release date (month, if before a holiday weekend, if during Christmas holiday, if during summer), star power (a score that accounts for actors, director, and producers), production budget, and if there the movie has some preexisting popularity, either through a novel/play adaptation or if there's a prequel. I'll also investigate other features such as runtime, rating, and franchise, but I have a hunch that these will have a smaller impact. 

In [ ]:
'''TO DOS:
- clean up genres and rating
- adjust budget for inflation'''

In [22]:
# storing
import pickle

# analysis 
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from datetime import datetime

from pprint import pprint

%matplotlib inline

First, let's load the movie data into a dataframe and check the quality of our data.

In [3]:
with open('pickled_data/all-movies-data.pkl', 'r') as picklefile:
    all_movies_data = pickle.load(picklefile)

In [144]:
# with open('pickled_data/failed-urls.pkl', 'r') as picklefile:
#     failed_urls = pickle.load(picklefile)

In [145]:
# print failed_urls

['http://www.boxofficemojo.com/movies/?id=romero.htm&adjust_yr=2016']


In [4]:
movies = pd.DataFrame(all_movies_data)

In [5]:
movies.shape

(16100, 19)

In [6]:
movies.head()

1-title 2-release_date 3-closing_date  \
0                    The A-Team     2010-06-11     2010-09-16   
1                      A.C.O.D.     2013-10-04     2013-11-07   
2  A.I. Artificial Intelligence     2001-06-29            NaT   
3                   Aaja Nachle     2007-11-30     2007-12-20   
4                     Aarakshan     2011-08-12     2011-09-22   

                                              actors     budget  \
0  [Liam Neeson, Bradley Cooper, Sharlto Copley, ...  110000000   
1  [Adam Scott, Catherine O'Hara, Richard Jenkins...        NaN   
2  [Haley Joel Osment, Frances O'Connor, Jude Law...  100000000   
3                                                           NaN   
4                                                           NaN   

             director            distributor  dom_total_gross  \
0      [Joe Carnahan]                    Fox         77222099   
1                            The Film Arcade           175705   
2  [Steven Spielberg]           Warner Bros.         78616689   
3                                   Yash Raj           484108   
4                      Reliance Big Pictures           651096   

   domestic_total_adj_gross    genre  intl_total_gross  oscar_noms  \
0                  84507200   Action         177238796           0   
1                    183100   Comedy               NaN           0   
2                 120841900   Sci-Fi         235926552           2   
3                    612200  Foreign           6773493           0   
4                    713400  Foreign            651096           0   

   oscar_wins                               producers   rating  runtime_mins  \
0           0  [Ridley Scott, Tony Scott, Alex Young]    PG-13           117   
1           0                      [Teddy Schwarzman]    PG-13            88   
2           0    [Kathleen Kennedy, Steven Spielberg]    PG-13           145   
3           0                                          Unrated           145   
4           0                                          Unrated           NaN   

   theaters             url       writers  
0      3544       ateam.htm  [Skip Woods]  
1        42        acod.htm                
2      3242          ai.htm                
3        66  aajanachle.htm                
4        91   aarakshan.htm

Time to do some cleaning...

In [7]:
# rename cols
movies.rename(columns = {'1-title': 'title', 
                         '2-release_date': 'release_date', 
                         '3-closing_date': 'closing_date'}, inplace = True)

In [172]:
# split actor, director, producer, writer columns
def get_person(peoplelist, i):
    '''Take a list of people and inde; return person at that index. Return blank if no list present.'''
    try:
        return peoplelist[i]
    except:
        return ''

def sep_people(df, person_type, num_people):
    '''For each column with list of people, split into n columns with one person in each column.
    Name new columns as person_type + n.
    
    Args:
    df -- name of dataframe
    person_type -- name of column (str)
    num_people -- number of columns to generate
    '''
    for n in range(num_people):
        df.loc[:, person_type + str(n+1)] = df.loc[:, person_type].apply(get_person, args=(n,))

# add seasons; rationale from http://www.boxofficemojo.com/seasonal/?chart=byseason
def get_season(date):
    '''Return season given month as int.'''
    months = (range(1, 13))
    seasons = ['winter', 'winter', 'spring', 'spring', 
           'summer', 'summer', 'summer', 'summer', 
           'fall', 'fall', 'holiday', 'holiday']
    seasons = dict(zip(months, seasons))
    try:
        return seasons[date.month]
    except:
        return ''
       
def add_season(df):
    '''Add corresponding season in new column based on release date.'''
    df['season'] = df.loc[:, 'release_date'].apply(get_season) 

# map raw genres to simple genres
def get_simple_genre(g):
    '''Recategorize raw genre into simple genre.'''
    raw_genres = tuple(by_genre.count().index)
    simple_genres = ['Act_Adven', 'Act_Adven', 'Act_Adven', 'Act_Adven', 'Act_Adven', 
                     'Act_Adven', 'Act_Adven', 'Act_Adven', 'Act_Adven', 'Act_Adven', 
                     'Animation', 'Comedy', 'Comedy', 'Comedy', 'Other', 'Crime_Thriller', 
                     'Crime_Thriller', 'Crime_Thriller', 'Crime_Thriller', 'Documentary', 
                     'Drama', 'Drama', 'Family', 'Family', 'Family', 'Fantasy_Scifi', 
                     'Fantasy_Scifi', 'Fantasy_Scifi', 'Foreign', 'Foreign', 'Foreign', 
                     'Historical', 'Historical', 'Horror', 'Horror', 'Horror', 'Other', 
                     'Other', 'Other', '', 'Historical', 'Historical', 'Historical', 
                     'Historical', 'Historical', 'Romance', 'Romance', 'Romance', 
                     'Romance', 'Fantasy_Scifi', 'Fantasy_Scifi', 'Fantasy_Scifi', 
                     'Fantasy_Scifi', 'Fantasy_Scifi', 'Fantasy_Scifi', 'Fantasy_Scifi', 
                     'Other', 'Other', 'Other', 'Other', 'Crime_Thriller', '', 'Other', 
                     'Other', 'Other', 'Other', 'Other', ]
    genres = dict(zip(raw_genres, simple_genres))
    try:
        return genres[g]
    except:
        return ''

def add_simple_genre(df):
    df['new_genre'] = df.loc[:, 'genre'].apply(get_simple_genre)
    
# clean up ratings
def get_clean_rating(r):
    rm_ratings = ['Unknown', 'Unrated', 'None', 'Not Yet Rated']
    return '' if r in rm_ratings else r

def clean_rating(df):
    df['clean_rating'] = df.loc[:, 'rating'].apply(get_clean_rating)
    
# add roi
def add_roi(df):
    df['dom_roi'] = df.loc[:, 'dom_total_gross'] / df.loc[:, 'budget']
    df['intl_roi'] = df.loc[:, 'intl_total_gross'] / df.loc[:, 'budget']
    
# add total time in theaters - not necessary if we have theaters?
def add_in_theaters_time(df):
    try:
        df['time_in_theaters'] = df.loc[:, 'closing_date'] - df.loc[:, 'release_date']
    except:
        df['time_in_theaters'] = ''
        


In [85]:
sep_people(movies, 'director', 3)
sep_people(movies, 'actors', 5)
sep_people(movies, 'writers', 2)
sep_people(movies, 'producers', 3)
add_season(movies)
add_simple_genre(movies) # try not to rerun this guy, takes a while
clean_rating(movies)

In [ ]:
### ONLY APPLICABLE TO MOVIES WITH BUDGETS

# adjust to 2015 dollars 
def get_infl_rate(startyear, endyear):
    '''Return inflation rate to 2015 dollars. 
    Args:
    startyear (float) -- start year
    endyear (int) -- end year
    Returns:
    infl (float) -- inflation rate
    '''
    try:
        cpiend = float(cpi.loc[endyear, 'AVE.'])
        cpistart = float(cpi.loc[int(startyear), 'AVE.'])
        infl = ((cpiend - cpistart) / cpistart) 
        return (1.0 + infl)
    except:
        return float('NaN')
    
def add_infl_rate(df):
    # add release year
    try:
        df['release_year'] = df['release_date'].apply(lambda x: x.year)
    except:
        df['release_year'] = None
    df['infl_rate'] = df['release_year'].apply(get_infl_rate, args=(2015,))
    
def add_adj_budget(df):
    df['budget_adj'] = df.loc[:, 'budget'] * df.loc[:, 'infl_rate']
    df['budget_adj'] = df['budget_adj'].fillna(0.0).astype(int)

# add roi calculations
def get_dom_roi(df):
    df['dom_roi'] = (df['dom_total_gross'] - df['budget'])/ df['budget']

def get_intl_roi(df):
    df['intl_roi'] = (df['intl_total_gross'] - df['budget'])/ df['budget']

In [ ]:
add_infl_rate(movies)
add_adj_budget(movies)
get_dom_roi(movies)
get_infl_rate(movies)

In [385]:
movies.head()

title release_date closing_date  \
0                    The A-Team   2010-06-11   2010-09-16   
1                      A.C.O.D.   2013-10-04   2013-11-07   
2  A.I. Artificial Intelligence   2001-06-29          NaT   
3                   Aaja Nachle   2007-11-30   2007-12-20   
4                     Aarakshan   2011-08-12   2011-09-22   

                                              actors     budget  \
0  [Liam Neeson, Bradley Cooper, Sharlto Copley, ...  110000000   
1  [Adam Scott, Catherine O'Hara, Richard Jenkins...        NaN   
2  [Haley Joel Osment, Frances O'Connor, Jude Law...  100000000   
3                                                           NaN   
4                                                           NaN   

             director            distributor  dom_total_gross  \
0      [Joe Carnahan]                    Fox         77222099   
1                            The Film Arcade           175705   
2  [Steven Spielberg]           Warner Bros.         78616689   
3                                   Yash Raj           484108   
4                      Reliance Big Pictures           651096   

   domestic_total_adj_gross    genre    ...           producers2  producers3  \
0                  84507200   Action    ...           Tony Scott  Alex Young   
1                    183100   Comedy    ...                                    
2                 120841900   Sci-Fi    ...     Steven Spielberg               
3                    612200  Foreign    ...                                    
4                    713400  Foreign    ...                                    

    season      new_genre clean_rating  release_year  infl_rate budget_adj  \
0   summer      Act_Adven        PG-13          2010   1.086955  119565019   
1     fall         Comedy        PG-13          2013   1.017428          0   
2   summer  Fantasy_Scifi        PG-13          2001   1.338323  133832298   
3  holiday        Foreign                       2007   1.143121          0   
4   summer        Foreign                       2011   1.053695          0   

    dom_roi  intl_roi  
0 -0.297981  0.611262  
1       NaN       NaN  
2 -0.213833  1.359266  
3       NaN       NaN  
4       NaN       NaN  

[5 rows x 40 columns]

#### cpi workspace

In [178]:
cpi = pd.read_csv('cpi-data.csv')

In [181]:
cpi.set_index('YEAR', inplace = True)

In [182]:
cpi.head()

AVE.
YEAR         
2016         
2015  237.017
2014  236.736
2013  232.957
2012  229.594

In [350]:
# def get_infl_rate(startyear, endyear):
#     '''Return inflation rate to 2015 dollars. 
#     Args:
#     startyear (int) -- start year
#     endyear (int) -- end year
#     Returns:
#     infl (float) -- inflation rate
#     '''
#     cpiend = float(cpi.loc[endyear, 'AVE.'])
#     cpistart = float(cpi.loc[startyear, 'AVE.'])
#     infl = ((cpiend - cpistart) / cpistart) 
#     return infl

In [334]:
type(float('NaN'))

float

In [353]:
movies.head()

title release_date closing_date  \
0                    The A-Team   2010-06-11   2010-09-16   
1                      A.C.O.D.   2013-10-04   2013-11-07   
2  A.I. Artificial Intelligence   2001-06-29          NaT   
3                   Aaja Nachle   2007-11-30   2007-12-20   
4                     Aarakshan   2011-08-12   2011-09-22   

                                              actors     budget  \
0  [Liam Neeson, Bradley Cooper, Sharlto Copley, ...  110000000   
1  [Adam Scott, Catherine O'Hara, Richard Jenkins...        NaN   
2  [Haley Joel Osment, Frances O'Connor, Jude Law...  100000000   
3                                                           NaN   
4                                                           NaN   

             director            distributor  dom_total_gross  \
0      [Joe Carnahan]                    Fox         77222099   
1                            The Film Arcade           175705   
2  [Steven Spielberg]           Warner Bros.         78616689   
3                                   Yash Raj           484108   
4                      Reliance Big Pictures           651096   

   domestic_total_adj_gross    genre    ...      writers2        producers1  \
0                  84507200   Action    ...                    Ridley Scott   
1                    183100   Comedy    ...                Teddy Schwarzman   
2                 120841900   Sci-Fi    ...                Kathleen Kennedy   
3                    612200  Foreign    ...                                   
4                    713400  Foreign    ...                                   

         producers2  producers3   season      new_genre  clean_rating  \
0        Tony Scott  Alex Young   summer      Act_Adven         PG-13   
1                                   fall         Comedy         PG-13   
2  Steven Spielberg               summer  Fantasy_Scifi         PG-13   
3                                holiday        Foreign                 
4                                 summer        Foreign                 

  release_year infl_rate budget_adj  
0         2010  1.086955  165000000  
1         2013  1.017428        NaN  
2         2001  1.338323  150000000  
3         2007  1.143121        NaN  
4         2011  1.053695        NaN  

[5 rows x 38 columns]

In [ ]:
# def get_adj_budget(budget):
#     adj_budget = 

In [377]:
add_adj_budget(movies)

In [378]:
movies.loc[0, 'budget_adj']

119565019

In [379]:
movies.head()

title release_date closing_date  \
0                    The A-Team   2010-06-11   2010-09-16   
1                      A.C.O.D.   2013-10-04   2013-11-07   
2  A.I. Artificial Intelligence   2001-06-29          NaT   
3                   Aaja Nachle   2007-11-30   2007-12-20   
4                     Aarakshan   2011-08-12   2011-09-22   

                                              actors     budget  \
0  [Liam Neeson, Bradley Cooper, Sharlto Copley, ...  110000000   
1  [Adam Scott, Catherine O'Hara, Richard Jenkins...        NaN   
2  [Haley Joel Osment, Frances O'Connor, Jude Law...  100000000   
3                                                           NaN   
4                                                           NaN   

             director            distributor  dom_total_gross  \
0      [Joe Carnahan]                    Fox         77222099   
1                            The Film Arcade           175705   
2  [Steven Spielberg]           Warner Bros.         78616689   
3                                   Yash Raj           484108   
4                      Reliance Big Pictures           651096   

   domestic_total_adj_gross    genre    ...      writers2        producers1  \
0                  84507200   Action    ...                    Ridley Scott   
1                    183100   Comedy    ...                Teddy Schwarzman   
2                 120841900   Sci-Fi    ...                Kathleen Kennedy   
3                    612200  Foreign    ...                                   
4                    713400  Foreign    ...                                   

         producers2  producers3   season      new_genre  clean_rating  \
0        Tony Scott  Alex Young   summer      Act_Adven         PG-13   
1                                   fall         Comedy         PG-13   
2  Steven Spielberg               summer  Fantasy_Scifi         PG-13   
3                                holiday        Foreign                 
4                                 summer        Foreign                 

  release_year infl_rate budget_adj  
0         2010  1.086955  119565019  
1         2013  1.017428          0  
2         2001  1.338323  133832298  
3         2007  1.143121          0  
4         2011  1.053695          0  

[5 rows x 38 columns]

In [322]:
type(movies.loc[0, 'infl_rate'])

float

In [321]:
movies.head(15)

title release_date closing_date  \
0                              The A-Team   2010-06-11   2010-09-16   
1                                A.C.O.D.   2013-10-04   2013-11-07   
2            A.I. Artificial Intelligence   2001-06-29          NaT   
3                             Aaja Nachle   2007-11-30   2007-12-20   
4                               Aarakshan   2011-08-12   2011-09-22   
5                               Aashayein   2010-08-27          NaT   
6   Aatsinki: The Story of Arctic Cowboys   2014-01-24   2014-03-06   
7                                 Abandon   2002-10-18   2002-12-19   
8                           The Abandoned   2007-02-23   2007-03-08   
9                              ABC Africa   2002-05-03   2002-08-29   
10                     The ABC's of Death   2013-03-08          NaT   
11                                   ABCD   2001-11-30          NaT   
12                                   ABCD   2013-02-08   2013-03-07   
13                                   ABCD   2015-06-19   2015-07-16   
14                        ABCs of Death 2   2014-10-31   2014-11-27   

                                               actors     budget  \
0   [Liam Neeson, Bradley Cooper, Sharlto Copley, ...  110000000   
1   [Adam Scott, Catherine O'Hara, Richard Jenkins...        NaN   
2   [Haley Joel Osment, Frances O'Connor, Jude Law...  100000000   
3                                                            NaN   
4                                                            NaN   
5                                                            NaN   
6                                                            NaN   
7     [Katie Holmes, Benjamin Bratt, Zooey Deschanel]   25000000   
8                                                            NaN   
9                                                            NaN   
10                                                           NaN   
11                                                           NaN   
12                                                           NaN   
13      [Varun Dhawan, Prabhu Dheva, Shraddha Kapoor]        NaN   
14                                                           NaN   

              director            distributor  dom_total_gross  \
0       [Joe Carnahan]                    Fox         77222099   
1                             The Film Arcade           175705   
2   [Steven Spielberg]           Warner Bros.         78616689   
3                                    Yash Raj           484108   
4                       Reliance Big Pictures           651096   
5                       Reliance Big Pictures              NaN   
6                              Argot Pictures             6601   
7                                   Paramount         10727683   
8                            After Dark Films          1331137   
9                                  New Yorker             9075   
10                                   Magnolia            21832   
11                                       Eros            83372   
12                         UTV Communications           222098   
13      [Remo D'Souza]     UTV Communications           892133   
14                                   Magnolia             7171   

    domestic_total_adj_gross        genre    ...      writers2  \
0                   84507200       Action    ...                 
1                     183100       Comedy    ...                 
2                  120841900       Sci-Fi    ...                 
3                     612200      Foreign    ...                 
4                     713400      Foreign    ...                 
5                        NaN      Foreign    ...                 
6                       7200  Documentary    ...                 
7                   16063800     Thriller    ...                 
8                    1683300       Horror    ...                 
9                      13600  Documentary    ...                 
10                     2390

In [235]:
print get_infl_rate(2012, 2015)

0.0323309842592


In [223]:
total = 10.0 * (1.0 + get_infl_rate(2012))
total

10.323309842591705

#### begin workspace for testing genre categorization

In [18]:
# original genre categories
raw_genres = tuple(by_genre.count().index)
print raw_genres

('Action', 'Action / Adventure', 'Action / Crime', 'Action Comedy', 'Action Drama', 'Action Fantasy', 'Action Horror', 'Action Thriller', 'Adventure', 'Adventure Comedy', 'Animation', 'Comedy', 'Comedy / Drama', 'Comedy Thriller', 'Concert', 'Crime', 'Crime Comedy', 'Crime Drama', 'Crime Thriller', 'Documentary', 'Drama', 'Drama / Thriller', 'Family', 'Family Adventure', 'Family Comedy', 'Fantasy', 'Fantasy Comedy', 'Fantasy Drama', 'Foreign', 'Foreign / Action', 'Foreign / Horror', 'Historical Drama', 'Historical Epic', 'Horror', 'Horror Comedy', 'Horror Thriller', 'IMAX', 'Music Drama', 'Musical', 'None', 'Period Action', 'Period Adventure', 'Period Comedy', 'Period Drama', 'Period Horror', 'Romance', 'Romantic Adventure', 'Romantic Comedy', 'Romantic Thriller', 'Sci-Fi', 'Sci-Fi Action', 'Sci-Fi Adventure', 'Sci-Fi Comedy', 'Sci-Fi Fantasy', 'Sci-Fi Horror', 'Sci-Fi Thriller', 'Sports', 'Sports Action', 'Sports Comedy', 'Sports Drama', 'Thriller', 'Unknown', 'War', 'War Drama', 'War

In [33]:
print len(raw_genres)

67


In [78]:
# code simple genres 
simple_genres = ['Act_Adven', 'Act_Adven', 'Act_Adven', 'Act_Adven', 'Act_Adven', 'Act_Adven', 
                 'Act_Adven', 'Act_Adven', 'Act_Adven', 'Act_Adven', 
                 'Animation', 'Comedy', 'Comedy', 'Comedy', 'Other', 
                 'Crime_Thriller', 'Crime_Thriller', 'Crime_Thriller', 'Crime_Thriller', 
                 'Documentary', 'Drama', 'Drama', 'Family', 'Family', 'Family', 'Fantasy_Scifi', 'Fantasy_Scifi', 
                 'Fantasy_Scifi', 'Foreign', 'Foreign', 'Foreign', 'Historical', 'Historical', 
                 'Horror', 'Horror', 'Horror', 'Other', 'Other', 'Other', '', 
                 'Historical', 'Historical', 'Historical', 'Historical', 'Historical', 
                 'Romance', 'Romance', 'Romance', 'Romance', 'Fantasy_Scifi', 'Fantasy_Scifi', 
                 'Fantasy_Scifi', 'Fantasy_Scifi', 'Fantasy_Scifi', 'Fantasy_Scifi', 
                 'Fantasy_Scifi', 'Other', 'Other', 'Other', 'Other', 'Crime_Thriller', 
                 '', 'Other', 'Other', 'Other', 'Other', 'Other', ]

In [79]:
# count how many genres
print set(simple_genres)
print len(set(simple_genres))

set(['', 'Romance', 'Family', 'Horror', 'Other', 'Act_Adven', 'Foreign', 'Drama', 'Animation', 'Historical', 'Crime_Thriller', 'Fantasy_Scifi', 'Comedy', 'Documentary'])
14


In [80]:
genres = dict(zip(raw_genres, simple_genres))

In [81]:
pprint(genres)

{'Action': 'Act_Adven',
 'Action / Adventure': 'Act_Adven',
 'Action / Crime': 'Act_Adven',
 'Action Comedy': 'Act_Adven',
 'Action Drama': 'Act_Adven',
 'Action Fantasy': 'Act_Adven',
 'Action Horror': 'Act_Adven',
 'Action Thriller': 'Act_Adven',
 'Adventure': 'Act_Adven',
 'Adventure Comedy': 'Act_Adven',
 'Animation': 'Animation',
 'Comedy': 'Comedy',
 'Comedy / Drama': 'Comedy',
 'Comedy Thriller': 'Comedy',
 'Concert': 'Other',
 'Crime': 'Crime_Thriller',
 'Crime Comedy': 'Crime_Thriller',
 'Crime Drama': 'Crime_Thriller',
 'Crime Thriller': 'Crime_Thriller',
 'Documentary': 'Documentary',
 'Drama': 'Drama',
 'Drama / Thriller': 'Drama',
 'Family': 'Family',
 'Family Adventure': 'Family',
 'Family Comedy': 'Family',
 'Fantasy': 'Fantasy_Scifi',
 'Fantasy Comedy': 'Fantasy_Scifi',
 'Fantasy Drama': 'Fantasy_Scifi',
 'Foreign': 'Foreign',
 'Foreign / Action': 'Foreign',
 'Foreign / Horror': 'Foreign',
 'Historical Drama': 'Historical',
 'Historical Epic': 'Historical',
 'Horror': '

#### </ end genre testing workspace >

Since we're interested in budget data, let's check to see how many movies of the 16,100 total movies have budget data.

In [150]:
all_movies.budget.count()

2874

Only 2,874 out of 16,100 movies have budget data. That's only 17.9% of the entire dataset! Let's take a closer look at movies with budget data.

In [61]:
# drop movies without budget data 
only_budget = all_movies[pd.notnull(all_movies['budget'])]

# add some roi calculations
only_budget['dom_roi'] = only_budget['dom_total_gross'] / only_budget['budget']
only_budget['intl_roi'] = only_budget['intl_total_gross'] / only_budget['budget']

/anaconda/lib/python2.7/site-packages/ipykernel/__main__.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
/anaconda/lib/python2.7/site-packages/ipykernel/__main__.py:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


In [65]:
only_budget.sort_values(by='dom_roi', ascending=False).head()

1-title 2-release_date 3-closing_date  \
9634          My Big Fat Greek Wedding     2002-04-19     2003-04-13   
12187                              Saw     2004-10-29     2004-12-30   
12871  Snow White and the Seven Dwarfs     1937-12-21            NaT   
7166                              Jaws     1975-06-20            NaT   
10316  One Flew Over the Cuckoo's Nest     1975-11-20            NaT   

                               actors   budget            director  \
9634     [John Corbett, Nia Vardalos]  5000000                None   
12187      [Cary Elwes, Danny Glover]  1200000         [James Wan]   
12871                            None  1499000                None   
7166                             None  7000000  [Steven Spielberg]   
10316  [Jack Nicholson, Danny DeVito]  3000000                None   

          distributor  dom_total_gross            genre  intl_total_gross  \
9634              IFC        241438208  Romantic Comedy         368744044   
12187      Lions Gate         55185045           Horror         103911669   
12871          Disney         66596803        Animation               NaN   
7166        Universal        260000000  Horror Thriller         470653000   
10316  United Artists        108981275            Drama               NaN   

       oscar_noms  oscar_wins                                producers rating  \
9634            1           0  [Paul Brooks, Gary Goetzman, Tom Hanks]     PG   
12187           0           0                                     None      R   
12871           0           0                                     None      G   
7166            0           0                                     None     PG   
10316           0           0                        [Michael Douglas]      R   

       runtime_(mins)  theaters                            url  \
9634               95      2016       mybigfatgreekwedding.htm   
12187             100      2467                        saw.htm   
12871              83       NaN                  snowwhite.htm   
7166              125       675                       jaws.htm   
10316             133       NaN  oneflewoverthecuckoosnest.htm   

                           writers    dom_roi   intl_roi  
9634                [Nia Vardalos]  48.287642  73.748809  
12187  [James Wan, Leigh Whannell]  45.987538  86.593058  
12871                         None  44.427487        NaN  
7166                          None  37.142857  67.236143  
10316                         None  36.327092        NaN

Let's look at only low budget films, which we're defining as films under $10m and evaluate the quality of the data.

In [151]:
# get only low budget (<$10m) films
low_budget = only_budget[only_budget['budget'] < 10000000]
low_budget.shape

(469, 20)

In [76]:
# pickle for later/challenge use
with open('pickled_data/low-budget-movies.pkl', 'w') as picklefile: 
    pickle.dump(low_budget, picklefile)

Looks like there are 469 total movies in our low budget films dataset. Some of these go back to 1925. Let's only look at movies since 1980.

In [99]:
newer_movies = low_budget[low_budget['2-release_date'] > datetime.strptime('1990', '%Y')]
newer_movies.count()

1-title             376
2-release_date      376
3-closing_date      283
actors              274
budget              376
director            205
distributor         376
dom_total_gross     372
genre               376
intl_total_gross    254
oscar_noms          376
oscar_wins          376
producers           159
rating              376
runtime_(mins)      372
theaters            370
url                 376
writers             145
dom_roi             372
intl_roi            254
dtype: int64

In [70]:
# top 10 movies by oscar nominations
low_budget.sort_values(by='oscar_noms', ascending=False).head(10)

1-title 2-release_date 3-closing_date  \
4482                        Fargo     1996-03-08            NaT   
11936          A Room with a View     1986-03-07            NaT   
10941                     Platoon     1986-12-19            NaT   
11268                Pulp Fiction     1994-10-14            NaT   
2976                        Crash     2005-05-06     2006-04-06   
5513   Good Night, and Good Luck.     2005-10-07     2006-03-16   
1793                      Boyhood     2014-07-11     2015-03-22   
9184             Midnight Express     1978-10-27            NaT   
3145           Dallas Buyers Club     2013-11-01     2014-05-01   
472                         Amour     2012-12-19     2013-05-16   

                                                  actors   budget  \
4482   [Steve Buscemi, William H. Macy, Frances McDor...  7000000   
11936  [Helena Bonham Carter, Daniel Day-Lewis, Judi ...  3000000   
10941  [Tom Berenger, Willem Dafoe, Charlie Sheen, Fo...  6000000   
11268  [John Travolta, Uma Thurman, Samuel L. Jackson...  8000000   
2976   [Sandra Bullock, Don Cheadle, Keith David, Lor...  6500000   
5513   [David Strathairn, Patricia Clarkson, George C...  7000000   
1793                    [Ethan Hawke, Patricia Arquette]  4000000   
9184                            [John Hurt, Randy Quaid]  2300000   
3145   [Matthew McConaughey, Jared Leto, Jennifer Gar...  5000000   
472                                                 None  8900000   

                  director         distributor  dom_total_gross  \
4482           [Joel Coen]            Gramercy         24611975   
11936        [James Ivory]             Cinecom         20966644   
10941       [Oliver Stone]      Orion Pictures        138530565   
11268  [Quentin Tarantino]             Miramax        107928762   
2976         [Paul Haggis]          Lions Gate         54580300   
5513      [George Clooney]  Warner Independent         31558003   
1793   [Richard Linklater]                 IFC         25352281   
9184                  None            Columbia         35000000   
3145    [Jean-Marc Vallee]      Focus Features         27298285   
472       [Michael Haneke]       Sony Classics          6739492   

                  genre  intl_total_gross  oscar_noms  oscar_wins  \
4482       Crime Comedy          60611975           8           2   
11936           Romance               NaN           8           3   
10941               War               NaN           8           4   
11268             Crime         213928762           7           1   
2976              Drama          98410061           6           3   
5513       Period Drama          54641191           6           0   
1793              Drama          44495281           6           1   
9184   Drama / Thriller               NaN           6           2   
3145              Drama          55198285           6           3   
472             Foreign          19839492           5           1   

                                               producers   rating  \
4482               [Tim Bevan, Ethan Coen, Eric Fellner]        R   
11936                                               None  Unrated   
10941                                  [Arnold Kopelson]        R   
11268   [Lawrence Bender, Michael Shamberg, Stacey Sher]        R   
2976             [Paul Haggis, Cathy Schulman, Bob Yari]        R   
5513   [Jennifer Fox, Samuel Hadida, Grant Heslov, Je...       PG   
1793                     [Richard Linklater, John Sloss]        R   
9184                                                None        R   
3145                                                None        R   
472                                                 None    PG-13   

       runtime_(mins)  theaters                       url  \
4482               95       716                 fargo.htm   
11936             117       150         roomwithaview.htm   
10941             120      1564               platoon.htm   
11268             154     

In [71]:
# top 10 movies by dom total gross 
low_budget.sort_values(by='dom_total_gross', ascending=False).head(10)

1-title 2-release_date 3-closing_date  \
7166                              Jaws     1975-06-20            NaT   
9634          My Big Fat Greek Wedding     2002-04-19     2003-04-13   
5610                            Grease     1978-06-16            NaT   
13006               The Sound of Music     1965-03-02            NaT   
7359                              Juno     2007-12-05     2008-06-19   
10941                          Platoon     1986-12-19            NaT   
5415                     The Godfather     1972-03-15            NaT   
4384                   Fahrenheit 9/11     2004-06-23     2004-10-28   
8705                        Magic Mike     2012-06-29     2012-09-27   
10316  One Flew Over the Cuckoo's Nest     1975-11-20            NaT   

                                                  actors   budget  \
7166                                                None  7000000   
9634                        [John Corbett, Nia Vardalos]  5000000   
5610                 [John Travolta, Olivia Newton-John]  6000000   
13006  [Julie Andrews, Christopher Plummer, Eleanor P...  8199999   
7359   [Ellen Page, Michael Cera, Jennifer Garner, Ja...  7500000   
10941  [Tom Berenger, Willem Dafoe, Charlie Sheen, Fo...  6000000   
5415   [Marlon Brando, Al Pacino, James Caan, Robert ...  6000000   
4384                                                None  6000000   
8705   [Riley Keough, Channing Tatum, Alex Pettyfer, ...  7000000   
10316                     [Jack Nicholson, Danny DeVito]  3000000   

                  director      distributor  dom_total_gross            genre  \
7166    [Steven Spielberg]        Universal        260000000  Horror Thriller   
9634                  None              IFC        241438208  Romantic Comedy   
5610                  None        Paramount        159978870          Musical   
13006        [Robert Wise]              Fox        158671368          Musical   
7359       [Jason Reitman]  Fox Searchlight        143495265   Comedy / Drama   
10941       [Oliver Stone]   Orion Pictures        138530565              War   
5415                  None        Paramount        133698921      Crime Drama   
4384       [Michael Moore]       Lions Gate        119194771      Documentary   
8705   [Steven Soderbergh]     Warner Bros.        113721571   Comedy / Drama   
10316                 None   United Artists        108981275            Drama   

       intl_total_gross  oscar_noms  oscar_wins  \
7166          470653000           0           0   
9634          368744044           1           0   
5610          394955690           1           0   
13006               NaN           0           0   
7359          231411584           4           1   
10941               NaN           8           4   
5415          245066411           0           0   
4384          222446882           0           0   
8705          167221571           0           0   
10316               NaN           0           0   

                                               producers rating  \
7166                                                None     PG   
9634             [Paul Brooks, Gary Goetzman, Tom Hanks]     PG   
5610                                                None     PG   
13006                        [Robert Wise, Saul Chaplin]      G   
7359                                      [Mason Novick]  PG-13   
10941                                  [Arnold Kopelson]      R   
5415                                                None      R   
4384                                                None      R   
8705   [Reid Carolin, Gregory Jacobs, Channing Tatum,...      R   
10316                                  [Michael Douglas]      R   

       runtime_(mins)  theaters                            url  \
7166              125       675                       jaws.htm   
9634               95      2016       mybigfatgreekwedding.htm   
5610              110       862                     grease.htm   
13006             174 

In [72]:
# top 10 movies by intl total gross 
low_budget.sort_values(by='intl_total_gross', ascending=False).head(10)

1-title 2-release_date 3-closing_date  \
7166                         Jaws     1975-06-20            NaT   
5610                       Grease     1978-06-16            NaT   
9634     My Big Fat Greek Wedding     2002-04-19     2003-04-13   
5075               The Full Monty     1997-08-15     1998-05-14   
553                     Annabelle     2014-10-03     2014-12-18   
4919  Four Weddings and a Funeral     1994-03-11            NaT   
5415                The Godfather     1972-03-15            NaT   
7359                         Juno     2007-12-05     2008-06-19   
4384              Fahrenheit 9/11     2004-06-23     2004-10-28   
3598                Dirty Dancing     1987-08-21            NaT   

                                                 actors   budget  \
7166                                               None  7000000   
5610                [John Travolta, Olivia Newton-John]  6000000   
9634                       [John Corbett, Nia Vardalos]  5000000   
5075                                    [Tom Wilkinson]  3500000   
553                                     [Alfre Woodard]  6500000   
4919                                       [Hugh Grant]  4400000   
5415  [Marlon Brando, Al Pacino, James Caan, Robert ...  6000000   
7359  [Ellen Page, Michael Cera, Jennifer Garner, Ja...  7500000   
4384                                               None  6000000   
3598                    [Patrick Swayze, Jennifer Grey]  6000000   

                director              distributor  dom_total_gross  \
7166  [Steven Spielberg]                Universal        260000000   
5610                None                Paramount        159978870   
9634                None                      IFC        241438208   
5075                None          Fox Searchlight         45950122   
553   [John R. Leonetti]  Warner Bros. (New Line)         84273813   
4919       [Mike Newell]                 Gramercy         52700832   
5415                None                Paramount        133698921   
7359     [Jason Reitman]          Fox Searchlight        143495265   
4384     [Michael Moore]               Lions Gate        119194771   
3598                None                  Vestron         63446382   

                genre  intl_total_gross  oscar_noms  oscar_wins  \
7166  Horror Thriller         470653000           0           0   
5610          Musical         394955690           1           0   
9634  Romantic Comedy         368744044           1           0   
5075           Comedy         257938649           4           1   
553            Horror         256873813           0           0   
4919  Romantic Comedy         245700832           2           0   
5415      Crime Drama         245066411           0           0   
7359   Comedy / Drama         231411584           4           1   
4384      Documentary         222446882           0           0   
3598          Romance         213954274           1           1   

                                    producers rating  runtime_(mins)  \
7166                                     None     PG             125   
5610                                     None     PG             110   
9634  [Paul Brooks, Gary Goetzman, Tom Hanks]     PG              95   
5075                                     None      R              91   
553                 [Peter Safran, James Wan]      R              95   
4919                [Tim Bevan, Eric Fellner]      R             118   
5415                                     None      R             175   
7359                           [Mason Novick]  PG-13              91   
4384                                     None      R             116   
3598                                     None  PG-13             105   

      theaters                          url           writers    dom_roi  \
7166       675                     jaws.htm              None  37.142857   
5610       862                   grease.htm              None  26.663145   
9634     

In [82]:
# genre? 

by_genre = newer_movies.groupby('genre')

In [97]:
ls = []
for i in list(newer_movies['genre']):
    print i
    break
    if i[:6] == 'Action':
       ls.append('Action')
    else:
        ls.append(i)

Drama


In [ ]:
newer_movies['new_genre'] = ls

In [91]:
zip(list(by_genre.count().index),ls)

[('Action', 'Action'),
 ('Action / Adventure', 'Action'),
 ('Action / Crime', 'Action'),
 ('Action Comedy', 'Action'),
 ('Action Drama', 'Action'),
 ('Action Thriller', 'Action'),
 ('Adventure', 'Adventure'),
 ('Animation', 'Animation'),
 ('Comedy', 'Comedy'),
 ('Comedy / Drama', 'Comedy / Drama'),
 ('Comedy Thriller', 'Comedy Thriller'),
 ('Concert', 'Concert'),
 ('Crime', 'Crime'),
 ('Crime Comedy', 'Crime Comedy'),
 ('Crime Drama', 'Crime Drama'),
 ('Crime Thriller', 'Crime Thriller'),
 ('Documentary', 'Documentary'),
 ('Drama', 'Drama'),
 ('Drama / Thriller', 'Drama / Thriller'),
 ('Family', 'Family'),
 ('Fantasy', 'Fantasy'),
 ('Fantasy Comedy', 'Fantasy Comedy'),
 ('Fantasy Drama', 'Fantasy Drama'),
 ('Foreign', 'Foreign'),
 ('Historical Drama', 'Historical Drama'),
 ('Horror', 'Horror'),
 ('Horror Comedy', 'Horror Comedy'),
 ('Horror Thriller', 'Horror Thriller'),
 ('IMAX', 'IMAX'),
 ('Music Drama', 'Music Drama'),
 ('Musical', 'Musical'),
 ('Period Adventure', 'Period Adventure

In [93]:
pd.get_dummies(newer_movies, columns = ['genre']).head()

1-title 2-release_date 3-closing_date  \
123           Affliction     1998-12-30            NaT   
200            Airplane!     1980-06-27            NaT   
207  Akeelah and the Bee     2006-04-28     2006-07-20   
223     Albino Alligator     1997-01-17     1997-02-13   
289       All or Nothing     2002-10-25            NaT   

                                                actors   budget  \
123                                     [Willem Dafoe]  6000000   
200                                   [Leslie Nielsen]  3500000   
207  [Laurence Fishburne, Angela Bassett, Keke Palmer]  8000000   
223                     [Matt Dillon, Viggo Mortensen]  6000000   
289                  [Taraji P. Henson, Sally Hawkins]  9000000   

                                       director     distributor  \
123                             [Paul Schrader]      Lions Gate   
200  [Jim Abrahams, David Zucker, Jerry Zucker]       Paramount   
207                                        None       Lionsgate   
223                                        None         Miramax   
289                                [Mike Leigh]  United Artists   

     dom_total_gross  intl_total_gross  oscar_noms      ...        \
123          6330054               NaN           2      ...         
200         83453539               NaN           0      ...         
207         18848430          18948425           0      ...         
223           339379               NaN           0      ...         
289           201546            201546           0      ...         

     genre_Romantic Thriller genre_Sci-Fi Action genre_Sci-Fi Horror  \
123                        0                   0                   0   
200                        0                   0                   0   
207                        0                   0                   0   
223                        0                   0                   0   
289                        0                   0                   0   

     genre_Sci-Fi Thriller  genre_Sports Comedy genre_Sports Drama  \
123                      0                    0                  0   
200                      0                    0                  0   
207                      0                    0                  0   
223                      0                    0                  0   
289                      0                    0                  0   

    genre_Thriller  genre_Unknown  genre_War  genre_Western  
123              0              0          0              0  
200              0              0          0              0  
207              0              0          0              0  
223              0              1          0              0  
289              0              1          0              0  

[5 rows x 64 columns]